### PR TITLE
Add Two Bucket exercise

### PR DESCRIPTION
**Pull request by tofische**
Added a new Haskell practice exercise Two Bucket based on the available problem specification.

https://forum.exercism.org/t/add-two-bucket-exercise/9920

### CODE REVIEW

File: exercises/practice/two-bucket/.meta/examples/success-standard/src/TwoBucket.hs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+type Buckets = (Int, Int)
+-- Sequence of moves
+type Moves = [Buckets]
+-- Solution candidates
+type Candidates = [Moves]
+
+measure :: Buckets -> Int -> (Int, Buckets)
+measure capacity target =
+    if null solution
+    then (0, (0, 0))
```
Comment (ErikSchierboom) [inline exercises/practice/two-bucket/.meta/examples/success-standard/src/TwoBucket.hs:17]: Why not use `Maybe` and return `Nothing` here?

**Review by ErikSchierboom — COMMENTED** [review]
CC @MatthijsBlom for a Haskell review

File: exercises/practice/two-bucket/.docs/instructions.append.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# Hints
```
[the remark this answers is not in the document]
Comment (tofische) [reply exercises/practice/two-bucket/.docs/instructions.append.md:1]: This is not an implementation hint, but an explanation, why there is no such parameter like starting bucket. I've however changed the title from Hints to Note.

File: exercises/practice/two-bucket/.meta/examples/success-standard/src/TwoBucket.hs (hunk 1 above)
Comment (tofische) [reply exercises/practice/two-bucket/.meta/examples/success-standard/src/TwoBucket.hs]: Changed.

File: exercises/practice/two-bucket/.docs/instructions.append.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ErikSchierboom) [reply exercises/practice/two-bucket/.docs/instructions.append.md]: Ah okay, sorry/

**Comment by ErikSchierboom** [comment]
Great. @MatthijsBlom can check the Haskell side of things.

**Pull request merged**